head_first_design_patterns/pde-bakk/ch10_jsonparser/CMakeLists.txt

```diff
@@ -1,13 +1,29 @@
 cmake_minimum_required(VERSION 3.22)
 project(json_parser)
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wextra")
 
-file(GLOB_RECURSE SOURCE_FILES "srcs/*/*.cpp" "srcs/*.cpp")
+file(GLOB_RECURSE SOURCE_FILES "srcs/*.cpp")
 file(GLOB HEADER_FILES "include/*.hpp")
 
 
 add_library(json_parser ${SOURCE_FILES} ${HEADER_FILES})
-add_executable(test json_parser.a main.cpp)
```

Comment:
Normally you'd place a CMakeLists file in each subdirectory and define the executables/libraries there, then the top-level CMakeLists can add directories without caring about the inner structure of that directory... this mostly just makes it easy to manage large projects though but I thought I'd mention it :)